simulator/memory/argv_loader/argv_loader.h

```diff
@@ -0,0 +1,44 @@
+/**
+ * argv_loader.h: loader of arguments to guest memory
+ * @author Rustem Yunusov <yunusov.rr@phystech.edu>
+ * Copyright 2019 MIPT-MIPS iLab project
+ */
+
+#ifndef ARGV_LOADER_H
+#define ARGV_LOADER_H
+
+#include <infra/argv.h>
+#include <infra/byte.h>
+#include <infra/types.h>
+#include <memory/memory.h>
+
+template <typename T, Endian endian>
+class ArgvLoader
+{
+public:
+    explicit ArgvLoader( const char* const* argv, const char* const* envp = nullptr);
+
+    size_t load_to( const std::shared_ptr<FuncMemory>& mem, T addr = 0);
```

Comment:
And that is a bad move. Address is a 64 bit address, regardless of how memory is used.